src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -627,16 +606,18 @@ private DocumentSnapshot lookupByPathNoXn(
                 .document(encodePathAsFirestoreDocumentName(lookupPath));
 
         RuntimeException lastException = null;
-        for (int retryNum = 0; retryNum < LOOKUP_RETRIES; retryNum++) {
+        for (int retryNum = 0; retryNum < fireStoreUtils.getFirestoreRetries(); retryNum++) {
             logger.info("FirestoreDirectoryDao lookupByPathNoXn - iteration {}", retryNum);
             try {
                 ApiFuture<DocumentSnapshot> docSnapFuture = docRef.get();
                 return docSnapFuture.get();
             } catch (AbortedException | ExecutionException ex) {
                 lastException = fireStoreUtils.handleExecutionException(ex, "lookupByPathNoXn");
             }
-            TimeUnit.SECONDS.sleep(LOOKUP_WAIT_SECONDS);
+            final long retryWait = SLEEP_BASE_MILLISECONDS * Double.valueOf(Math.pow(2.5, retryNum)).longValue();
```

Comment:
Does this produce a different result than casting to `long`? Personally I find the cast easier to read.
```suggestion
            final long retryWait = (long) (SLEEP_BASE_MILLISECONDS * Math.pow(2.5, retryNum));
```

Reply:
With the following results, I think either would work just fine!
I ran this test:
  
    private static final int SLEEP_BASE_MILLISECONDS = 1000;

    @Test
    public void testRetryTime(){
        for(int retryNum = 0; retryNum < 10; retryNum++) {
            final long retryWait = SLEEP_BASE_MILLISECONDS * Double.valueOf(Math.pow(2.5, retryNum)).longValue();
            final long retryWait2 = (long) (SLEEP_BASE_MILLISECONDS * Math.pow(2.5, retryNum));
            logger.info("{} vs. {}", retryWait, retryWait2);
        }
    }
`

![image](https://user-images.githubusercontent.com/13254229/122982470-e95a4980-d368-11eb-9ac4-6f9cf940a9d5.png)

Reply:
When I convert to seconds, they produce the same result. So, I'll go with the one you sugguested!

Reply:
The results make sense, the original code is truncating `double` to `long` before the multiplication, and the new code truncates afterwards. You could keep the original behavior if you want by writing `SLEEP_BASE_MILLISECONDS * (long) Math.pow(2.5, retryNum)`